File: Tratamento de Exceções em Java e Python/Trabalho/Exemplos Praticos/Ex002.java

import java.io.*;

public class FileExceptionHandling {
    public static void main(String[] args) {
        File file = new File("example.txt");
        BufferedReader reader = null;
        try {
            reader = new BufferedReader(new FileReader(file));
            String line = null;
            while ((line = reader.readLine()) != null) {
                System.out.println(line);
            }
            throw new IOException("Simulated Error"); // Simula um erro durante a leitura
        } catch (FileNotFoundException e) {
            System.out.println("File not found: " + e.getMessage());
        } catch (IOException e) {
            System.out.println("IO Error: " + e.getMessage());
        } finally {
            try {
                if (reader != null) {
                    reader.close();
                }
            } catch (IOException e) {
                System.out.println("Failed to close reader: " + e.getMessage());
            }
        }
    }
}

File: Tratamento de Exceções em Java e Python/Trabalho/Exemplos Praticos/Ex002.py
def read_process_file():
    try:
        with open("example.txt", "r") as file:
            data = file.read()
            print(data)
        raise ValueError("Simulated Error")  # Simula um erro após a leitura
    except FileNotFoundError:
        print("The file was not found.")
    except IOError:
        print("An error occurred while reading or writing to the file.")
    except Exception as e:
        print("A general error occurred:", e)
    finally:
        print("Execution completed, whether an error occurred or not.")

read_process_file()
